client/src/components/_common/Paginator/Paginator.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { Button } from 'reactstrap';
+import PropTypes from 'prop-types';
+import './Paginator.module.scss';
+
+const PaginatorPage = ({ number, callback, current }) => {
+  return (
+    <div styleName="page-root">
+      <Button
+        styleName={`page ${number === current ? 'current' : ''}`}
+        onClick={() => callback(number)}
+      >
+        {number}
+      </Button>
+    </div>
+  );
+};
+
+PaginatorPage.propTypes = {
+  number: PropTypes.number.isRequired,
+  callback: PropTypes.func.isRequired,
+  current: PropTypes.number.isRequired
+};
+
+const Paginator = ({ pages, current, callback }) => {
+  let start, end;
+  if (pages === 1) {
+    end = 0;
+    start = 1;
+  } else if (pages > 2 && pages < 7) {
+    start = 2;
+    end = pages - 1;
+  } else if (pages > 7 && current <= 4) {
+    start = 2;
+    end = 6;
+  } else if (pages > 7 && current > pages - 5) {
+    start = pages - 5;
+    end = pages - 1;
+  } else {
+    start = current - 2;
+    end = current + 2;
+  }
```

Comment:
__Recommended__ (does not block approval)

I recommend that, if not now, then later, `7` be tied to a constant or—better yet—a prop, because:
- The value expected by [the design definition](https://confluence.tacc.utexas.edu/display/UP/UI+-+Components+-+Pagination) is actually `11`.
- Other use cases (like maybe Onboarding) could want a different value (like `7`).

Reply:
Added "spread" property

Reply:
![image](https://user-images.githubusercontent.com/17181582/109529278-32655580-7a7b-11eb-9898-01e60c8b84e5.png)